src/core/Chip/index.js

```diff
@@ -0,0 +1,69 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import Icon from '../Icon'
+import s from './styles'
+
+// Only allow lower case letters, optionally seperated by underscore
+const mdIconNamePattern = /^[a-z]+_*[a-z]+$/
+
+class Chip extends React.PureComponent {
+    onRemove = event => {
+        event.stopPropagation()
+        this.props.onRemove()
+    }
+
+    renderIcon(icon) {
+        return mdIconNamePattern.test(icon) ? (
+            <Icon name={icon} className={s('icon')} />
+        ) : (
+            <img src={icon} alt="chip-icon" className={s('image-icon')} />
+        )
+    }
+
+    render() {
+        const {
+            label,
+            onClick,
+            selected,
+            disabled,
+            dragging,
+            icon,
+            onRemove,
+        } = this.props
+
+        return (
+            <div
+                className={s('container', {
+                    static: !onClick,
+                    selected,
+                    disabled,
+                    dragging,
+                })}
+                onClick={(!disabled && onClick) || undefined}
```

Comment:
- simplify this to `onClick={onClick}`
- handle disabling button using css as you already are adding the disabled class to `.container`
- add `Chip.defaultProps` for `onClick: undefined`

Reply:
`.disabled` class does not have `pointer-events: none;` but `cursor: not-allowed;`. Both can't work together. Hence my current solution. A lot of components are using `cursor: not-allowed;` for disabled states (Button, Menu, LabelFields, etc), so if we want to change it here, maybe we should also visit other components to make sure we are consistent.

Reply:
What he means is that if we use this code:

```
.container.disabled {
    pointer-events: none;
    cursor: not-allowed;
}
```

Then when you hover the `.container`, you will not get the `cursor: not-allowed`, because no pointer events will be fired by the browser from the `.container` element.

Reply:
We can move the special logic to a function instead of having it inline.

b4b5314